### ISSUE TITLE
SD card example (sdspi) does not work with ESP32-S3-EYE (IDFGH-9495)

### ISSUE THREAD

**Issue by robosina** [espressif/esp-idf#10855]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

v4.4.4-148-g4c2afac355

### Operating System used.

Linux

### How did you build your project?

Command line with idf.py

### If you are using Windows, please specify command line type.

None

### Development Kit.

ESP32-S3-EYE 

### Power Supply used.

Battery

### What is the expected behavior?

Should mount the SD card and write something into it

### What is the actual behavior?

Upon running the example and opening the monitor using the idf.py script, I observe the following problem:

```
E (386) sdmmc_sd: sdmmc_init_sd_if_cond: send_if_cond (1) returned 0x108
E (386) vfs_fat_sdmmc: sdmmc_card_init failed (0x108).
I (386) gpio: GPIO[34]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0 
E (396) example: Failed to initialize the card (ESP_ERR_INVALID_RESPONSE). Make sure SD card lines have pull-up resistors in place.
```

Full Log --> [Logs](https://pastebin.com/wdJV23V0)

### Steps to reproduce.

Running the example without any modifications (although I did try the other solutions I saw in the issues, but none of them worked for me)

**Comment by igrr** [comment]
Hi @robosina,
Indeed it's not possible to use the SD card on ESP32-S3-EYE board in SPI mode. According to the [schematics](https://dl.espressif.com/dl/schematics/SCH_ESP32-S3-EYE-MB_20211201_V2.2.pdf), the card has 3 pins connected to the ESP: CMD, CLK, D0. Therefore it is only possible to use it in 1-bit SD mode. To be able to use the card in SPI mode, pin D3 would also have to be connected to the ESP.

**Comment by robosina** [comment]
@igrr , First of all, thanks for your response, but I have tried the 1 bit and it did not work. However, I re-tried it (I have configured it from the `menuconfig` and it is using the 1 bit mode) and it did not work and the same error appears. 

https://github.com/espressif/esp-idf/blob/master/examples/storage/sd_card/sdmmc/main/sd_card_example_main.c#L102

**Comment by igrr** [comment]
From your verbose log:

```
D (877) sdmmc_periph: using GPIO14 as clk pin
...
D (897) sdmmc_periph: using GPIO15 as cmd pin
...
D (907) sdmmc_periph: using GPIO2 as d0 pin
```

Could you please check the schematics of the board and your configuration again? The board uses different pins for SD card connection than the ones in your log.

**Comment by robosina** [comment]
@igrr , I appreciate your reply. You are correct about that since the code uses SDMMC_SLOT_CONFIG_DEFAULT, therefore it will not be possible to use that configuration for every board and we should tune it for our usage.

**Comment by igrr** [comment]
SDMMC_SLOT_CONFIG_DEFAULT shouldn't be a problem: the example actually configures the specific pins right here:
https://github.com/espressif/esp-idf/blob/acac972f70984e045e891c5edf95e0693f1b80a6/examples/storage/sd_card/sdmmc/main/sd_card_example_main.c#L107-L116
In the example the pin numbers (`CONFIG_EXAMPLE_PIN_*`) come from menuconfig. You can either customize the pins in menuconfig or change them right in the code. Please revisit this part of the example readme: https://github.com/espressif/esp-idf/tree/master/examples/storage/sd_card/sdmmc#pin-assignments-for-esp32-s3.

**Comment by igrr** [comment]
Oh, now I understand the issue. You are using IDF v4.4.x. The example uses `CONFIG_SOC_SDMMC_USE_GPIO_MATRIX` macro which was only added in IDF v5.x, so this code is never executed.

We'll fix the example in release/v4.4 branch.

**Comment by robosina** [comment]
@igrr , In menuconfig, I could not find a place to set the pin number (I have checked the SD/MMC Example configuration and there I can only define bus width and an option for formatting the card or not, perhaps it is related to the version I am running). As a temporary solution, perhaps I should define `CONFIG_SOC_SDMMC_USE_GPIO_MATRIX` in CMake as a target definition (perhaps some kind of device discovery would be better when we run the `idf.py set-target` command)

**Comment by igrr** [comment]
> In menuconfig, I could not find a place to set the pin number (I have checked the SD/MMC Example configuration in menuconfig and there I can only define:

Yes, that's because the example uses `SOC_SDMMC_USE_GPIO_MATRIX` in the Kconfig file as a condition for showing these pin settings. This is a bug, since this `SOC_SDMMC_USE_GPIO_MATRIX` is not defined in IDF v4.4.x releases in Kconfig. We missed this fact when backporting the example updates from the newer IDF v5.x release versions.

We will fix that, so that the pins will become configurable in menuconfig as intended.

In the meantime, you can simply write
```c++
    slot_config.clk = GPIO_NUM_39;
    slot_config.cmd = GPIO_NUM_38;
    slot_config.d0 = GPIO_NUM_40;
```
in your application without any ifdefs — it will work as long as you are using the ESP32-S3.

**Comment by robosina** [comment]
Yes, I see that now, I have done the same kind of work, thanks

**Comment by RathiSonika** [comment]
Fix is available in [release/v4.4](https://gitlab.espressif.cn:6688/espressif/esp-idf/-/tree/release/v4.4)
